### PR TITLE
Subscribe block

### PR DESCRIPTION
**Pull request by idubinskiy**

### CODE REVIEW

File: channel.go [hunk 1]
```diff
@@ -60,17 +63,71 @@ func (c *channel) IsSubscribed() bool {
...
+
+	go func() {
+		var err error
+		select {
+		case <-c.subscribeSuccess:
+			err = nil
+		case <-time.After(o.successTimeout):
+			err = ErrTimedOut
+		}
+		select {
```
Comment (timstudd) [inline channel.go]: Please add an empty line separating the two select statements

File: channel.go [hunk 2]
```diff
@@ -60,17 +63,71 @@ func (c *channel) IsSubscribed() bool {
...
+		case doneChan <- err:
+		default:
```
Comment (timstudd) [inline channel.go]: Please add a comment concerning the intended pass-through default behavior.  It's slightly unclear as it stands.

File: channel.go [hunk 3]
```diff
@@ -171,8 +251,18 @@ func (c *privateChannel) Subscribe() error {
...
-		return fmt.Errorf("Got non-200 status code from auth endpoint: %d", res.StatusCode)
+		var body []byte
+		var bodyStr string
+		body, err = ioutil.ReadAll(res.Body)
+		if err != nil {
+			bodyStr = fmt.Sprintf("Error reading response body: %s", err)
+		} else {
+			bodyStr = string(body)
+		}
+		return AuthError{
```
Comment (timstudd) [inline channel.go]: This is a nit, but I'd prefer a new line before the return block

File: channel.go [hunk 4]
```diff
@@ -60,17 +63,71 @@ func (c *channel) IsSubscribed() bool {
...
+// WithSuccessTimeout returns a SubscribeOption that sets the time that a subscription
+// request will wait for a success response from Pusher before timing out. The
+// default is 10 seconds.
+func WithSuccessTimeout(d time.Duration) SubscribeOption {
+	return func(o *subscribeOptions) {
+		o.successTimeout = d
+	}
+}
+
+// ErrTimedOut is the error returned when there is a timeour waiting for a subscription
```
Comment (melindabernrdo) [inline channel.go]: *timeout?

**Pull request merged**